cmd/thanos/compact.go

```diff
@@ -458,6 +460,56 @@ func runCompact(
 		return cleanPartialMarked()
 	}
 
+	if conf.compactionProgressMetrics {
+		g.Add(func() error {
+			unRegisterer := &receive.UnRegisterer{Registerer: reg}
+			ps := compact.NewCompactionProgressCalculator(unRegisterer, tsdbPlanner)
+			var ds *compact.DownsampleProgressCalculator
+			if !conf.disableDownsampling {
+				ds = compact.NewDownsampleProgressCalculator(unRegisterer)
+			}
+
+			return runutil.Repeat(5*time.Minute, ctx.Done(), func() error {
+
+				if err := sy.SyncMetas(ctx); err != nil {
+					return errors.Wrapf(err, "could not sync metas")
+				}
+
+				originalMetas := sy.Metas()
+				groups, err := grouper.Groups(originalMetas)
+				if err != nil {
+					return errors.Wrapf(err, "could not group original metadata")
+				}
+
+				if !conf.disableDownsampling {
+					downGroups, err := grouper.Groups(originalMetas)
+					if err != nil {
+						return errors.Wrapf(err, "could not group original metadata into downsample groups")
+					}
+					for _, group := range downGroups {
+						ds.DownsampleProgressMetrics.NumberOfBlocksDownsampled.WithLabelValues(group.Key())
+					}
+					if err := ds.ProgressCalculate(ctx, downGroups); err != nil {
+						return errors.Wrapf(err, "could not simulate downsampling")
+					}
+				}
+
+				for _, group := range groups {
+					ps.CompactProgressMetrics.NumberOfCompactionRuns.WithLabelValues(group.Key())
+					ps.CompactProgressMetrics.NumberOfCompactionBlocks.WithLabelValues(group.Key())
+				}
+
+				if err = ps.ProgressCalculate(ctx, groups); err != nil {
+					return errors.Wrapf(err, "could not simulate planning")
```

Comment:
Can you update the comment to be more specific? `could not calculate compaction progress`